packages/lodestar/src/chain/validation/syncCommittee.ts

```diff
@@ -0,0 +1,107 @@
+import {
+  CachedBeaconState,
+  computeSubnetsForSyncCommittee,
+  fast,
+  getIndicesInSubSyncCommittee,
+} from "@chainsafe/lodestar-beacon-state-transition";
+import {IBeaconConfig} from "@chainsafe/lodestar-config";
+import {altair} from "@chainsafe/lodestar-types";
+import {IBeaconDb} from "../../db";
+import {ISyncCommitteeJob, SyncCommitteeError, SyncCommitteeErrorCode} from "../errors/syncCommitteeError";
+import {IBeaconChain} from "../interface";
+
+export async function validateGossipSyncCommittee(
+  config: IBeaconConfig,
+  chain: IBeaconChain,
+  db: IBeaconDb,
+  syncCommitteeJob: ISyncCommitteeJob,
+  subnet: number
+): Promise<void> {
+  const {syncCommittee, validSignature} = syncCommitteeJob;
+  // The signature's slot is for the current slot
+  if (chain.clock.currentSlot < syncCommittee.slot) {
+    throw new SyncCommitteeError({
+      code: SyncCommitteeErrorCode.FUTURE_SLOT,
+      currentSlot: chain.clock.currentSlot,
+      syncCommitteeSlot: syncCommittee.slot,
+      job: syncCommitteeJob,
+    });
+  }
+
+  if (chain.clock.currentSlot > syncCommittee.slot) {
+    throw new SyncCommitteeError({
+      code: SyncCommitteeErrorCode.PAST_SLOT,
+      currentSlot: chain.clock.currentSlot,
+      syncCommitteeSlot: syncCommittee.slot,
+      job: syncCommitteeJob,
+    });
+  }
+
+  // The block being signed over has been seen
+  if (!chain.forkChoice.hasBlock(syncCommittee.beaconBlockRoot)) {
+    throw new SyncCommitteeError({
+      code: SyncCommitteeErrorCode.UNKNOWN_BEACON_BLOCK_ROOT,
+      beaconBlockRoot: syncCommittee.beaconBlockRoot as Uint8Array,
+      job: syncCommitteeJob,
+    });
+  }
+
+  // There has been no other valid sync committee signature with same slot and validatorIndex
+  if (db.seenSyncCommiteeCache.hasSyncCommitteeSignature(subnet, syncCommittee)) {
+    throw new SyncCommitteeError({
+      code: SyncCommitteeErrorCode.SYNC_COMMITTEE_ALREADY_KNOWN,
+      root: config.types.altair.SyncCommitteeSignature.hashTreeRoot(syncCommittee),
+      job: syncCommitteeJob,
+    });
+  }
+
+  const headState = chain.getHeadState();
```

Comment:
Note the conversation about forkChoice efficiency. Calling getHead() here potentially means 1000 times per slot

Reply:
in this scenario we don't really need head state since what we want to get validator pubkey and sync committees. I suppose we should create `chain.getHeadStateUnsafe()` and audit all use cases of `chain.getHeadState()` to use that instead?